xrspatial/zonal.py

```diff
@@ -106,105 +122,191 @@ def _stats(
                 raise ValueError(stats)
             stats_dict[stats].append(stats_func(zone_values))
 
-    unique_zones = list(map(_to_int, unique_zones))
     stats_dict["zone"] = unique_zones
 
     return stats_dict
 
 
-def _stats_dask(
-    zones: xr.DataArray,
-    values: xr.DataArray,
+@ngjit
+def _strides(flatten_zones, unique_zones):
+    num_elements = flatten_zones.shape[0]
+    num_zones = len(unique_zones)
+    strides = np.zeros(len(unique_zones), dtype=np.int32)
+
+    count = 0
+    for i in range(num_zones):
+        while (count < num_elements) and (
+                flatten_zones[count] == unique_zones[i]):
+            count += 1
+        strides[i] = count
+
+    return strides
+
+
+@delayed
+def _stats_func_dask_numpy(
+    zones_block: np.array,
+    values_block: np.array,
+    unique_zones: np.array,
+    zone_ids: np.array,
+    func: callable,
```

Comment:
I think `callable` should have a capital C so that it is the type rather than the function.